src/catalog/catalog_manager.py

```diff
@@ -0,0 +1,74 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from src.utils.configuration_manager import ConfigurationManager
+from src.utils.logging_manager import LoggingManager
+from src.utils.logging_manager import LoggingLevel
+
+from urllib.parse import urlparse
+
+from src.catalog.dataset_df import *
+
+CATALOG_DIR = "catalog"
+DATASET_FILE = "dataset"
+
+catalog = {}
+
+
+class CatalogManager(object):
+
+    _instance = None
+    _catalog = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(CatalogManager, cls).__new__(cls)
+
+            cls._instance.bootstrap_catalog()
+
+        return cls._instance
+
+    def bootstrap_catalog(self):
+
+        eva_dir = ConfigurationManager().get_value("core", "location")
+        output_url = os.path.join(eva_dir, CATALOG_DIR)
```

Comment:
Aren't catalog_dir_url and output_url the same? 

output_url is only being for logging.

Reply:
Cleaned up this code. 

There is an issue with converting file paths to urls like:
`/tmp/eva_datsets` to `file:///tmp/eva_datasets`

I am not sure if this is the correct solution (with urlparse).